qiskit_nature/second_q/hamiltonians/quadratic_hamiltonian.py

```diff
@@ -123,29 +123,40 @@ def __init__(
                 if not _is_antisymmetric(antisymmetric_part, rtol=rtol, atol=atol):
                     raise ValueError("Antisymmetric part must be antisymmetric.")
 
-        self._hermitian_part = hermitian_part
-        self._antisymmetric_part = antisymmetric_part
-        self._constant = constant
-
-        if self._hermitian_part is None:
-            self._hermitian_part = np.zeros((self._num_modes, self._num_modes))
-        if self._antisymmetric_part is None:
-            self._antisymmetric_part = np.zeros((self._num_modes, self._num_modes))
+        data: dict[str, float | np.ndarray] = {"": constant}
+        if hermitian_part is not None:
+            data["+-"] = hermitian_part
+        if antisymmetric_part is not None:
+            data["++"] = antisymmetric_part
+        super().__init__(data)
```

Comment:
You say, that this class implements the `TolerancesMixin` and allow `rtol` and `atol` as argument to `__init__`. However, you don't set the internal attributes to the provided values, which means that these will only be considered during the course of `__init__` and will then be lost.

Reply:
I deleted the `TolerancesMixin` specification. It's actually redundant because `PolynomialTensor` already inherits it. The `rtol` and `atol` arguments to `__init__` are only for the input validation.